src-cpp/CMakeLists.txt

```diff
@@ -37,49 +37,40 @@ if(NOT RDKAFKA_BUILD_STATIC)
 endif()
 
 # Generate pkg-config file
-set(PKG_CONFIG_NAME
-    "librdkafka++"
-)
-set(PKG_CONFIG_DESCRIPTION
-    "The Apache Kafka C/C++ library"
-)
-set(PKG_CONFIG_VERSION
-    "${PROJECT_VERSION}"
-)
-set(PKG_CONFIG_REQUIRES "rdkafka")
-set(PKG_CONFIG_CFLAGS
-    "-I\${includedir}"
-)
-set(PKG_CONFIG_LIBS
-    "-L\${libdir} -lrdkafka++"
-)
-set(PKG_CONFIG_LIBS_PRIVATE
-    "-lrdkafka"
-)
-configure_file(
+set(PKG_CONFIG_VERSION "${PROJECT_VERSION}")
+if(NOT RDKAFKA_BUILD_STATIC)
+  set(PKG_CONFIG_NAME "librdkafka++")
+  set(PKG_CONFIG_DESCRIPTION "The Apache Kafka C/C++ library")
+  set(PKG_CONFIG_REQUIRES "rdkafka")
+  set(PKG_CONFIG_CFLAGS "-I\${includedir}")
+  set(PKG_CONFIG_LIBS "-L\${libdir} -lrdkafka++")
+  set(PKG_CONFIG_LIBS_PRIVATE "-lrdkafka")
+  configure_file(
     "../packaging/cmake/rdkafka.pc.in"
     "${GENERATED_DIR}/rdkafka++.pc"
     @ONLY
-)
-install(FILES ${GENERATED_DIR}/rdkafka++.pc
-    DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig"
-)
-if(RDKAFKA_BUILD_STATIC)
-  set(PKG_CONFIG_NAME
-    "librdkafka++-static"
   )
-  set(PKG_CONFIG_DESCRIPTION
-    "The Apache Kafka C/C++ library (static)"
-  )
-  set(PKG_CONFIG_LIBS
-    "-L\${libdir} \${libdir}/librdkafka++.a"
+  install(
+    FILES ${GENERATED_DIR}/rdkafka++.pc
+    DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig"
   )
+else()
+  set(PKG_CONFIG_NAME "librdkafka++-static")
+  set(PKG_CONFIG_DESCRIPTION "The Apache Kafka C/C++ library (static)")
+  set(PKG_CONFIG_REQUIRES "")
+  set(PKG_CONFIG_CFLAGS "-I\${includedir} -DLIBRDKAFKA_STATICLIB")
+  if(WIN32)
```

Comment:
This is dependent on MINGW, not WIN32, in other places, why not here?

Reply:
The check in `src/CMakeLists.txt` also uses `WIN32` instead of `MINGW` (see https://github.com/neptoess/librdkafka/blob/add-mingw-to-packaging/src/CMakeLists.txt#L312-L319).

I can change these, but I'm not sure we're risking too much either way. `pkg-config` _does_ work for compilers other than MinGW, e.g. MSVC, clang, and these dependencies will still apply.

Reply:
You did make me realize there was a bug for static builds on non-Windows platforms, so I fixed that